.build_petsc_for_ci.sh

```diff
@@ -20,7 +20,7 @@ if test $BUILD_PETSC ; then
 	    --with-scalar-type=real \
 	    --with-shared-libraries=1 \
 	    --with-debugging=0 \
-	    {C,CXX,F}OPTFLAGS="-O3 -march=native" \
```

Comment:
One could use `mtune` if we 
1. Care about the PETSc CI performance (or think people might use this to build their own PETSc and care about performance).
2. Think that we're likely to get the same CPU type _most_ but not all of the time.

I doubt these are true, particularly 1, but perhaps this could inform the docs (as I think march came from the docs).

Reply:
I do agree :+1: 
`-mtune` would have been the better solution.